### ISSUE TITLE
search_lucene partly broken for user after upgrading to v8.1.1

### ISSUE THREAD

**Issue by mbressman** [owncloud/core#18571]
After upgrading to v8.1.1, I can still search my ownCloud install (under my user account, which is basically the only one on ownCloud right now) for text that is contained in the file name, but it no longer seems to be searching contents of certain file types. I upgraded and search_lucene was disabled, and I re-enabled it via the web interface (but also tried disabling and re-enabling via OCC when connecting to my server via SSH), but no go. Any thoughts or suggestions? Thanks!

P.S. I'm not very familiar with GitHub, so I apologize in advance if I'm leaving something out, or am improperly submitting this request.  I did post this over on the ownCloud forum, but it was suggested that I file a bug report here as well (https://forum.owncloud.org/viewtopic.php?f=31&t=29956).  Thanks!

Also - I wanted to try and see if I could somehow just delete the existing search index and let ownCloud re-create it, to see if that would fix the issue (since I know in Windows sometimes re-creating the search index can resolve issues), but wasn't sure what the proper procedure was to do that (i.e. just deleting the data under /owncloud/data/$user/lucene_index folder, or having to do something with the database also, or something else entirely), and didn't want to break anything further without first checking with the developers and those that know a lot more about this than I do :)

**Comment by MorrisJobke** [comment]
@mbressman Can you try owncloud/search_lucene#57 ?

cc @butonic

**Comment by mbressman** [comment]
MorrisJobke - thanks for the response.  I'm somewhat unfamiliar with GitHub, so just to be clear, you want me to try editing the 4 files with the information posted here (https://github.com/owncloud/search_lucene/pull/57/files), right?  And then after doing that, what should I do to presumably restart ownCloud, or I don't have to do anything?

Thanks!

**Comment by MorrisJobke** [comment]
> MorrisJobke - thanks for the response. I'm somewhat unfamiliar with GitHub, so just to be clear, you want me to try editing the 4 files with the information posted here (https://github.com/owncloud/search_lucene/pull/57/files), right? And then after doing that, what should I do to presumably restart ownCloud, or I don't have to do anything?

Restart shouldn't be needed. It should solve your issue. If this works, you could report back :)

**Comment by mbressman** [comment]
Ok, so just tried editing those 4 files, and then went back into my ownCloud via web interface and tried searching for some text that I know exists in some text files and was still getting no results found.  Any ideas?  Thanks!

**Comment by MorrisJobke** [comment]
@mbressman While in the browser: Can you press F12 (in Firefox and Chrome the web development tools should open). Click on the "Network" tab. Then enter the search term in the search box. Can you have a look at the requests, that are listed in the "Network" tab. Click on the one that showed up and maybe send a screenshot of the "reponse" tab (inside the just opened subwindow).

**Comment by mbressman** [comment]
![screenshot - f12 network tab](https://cloud.githubusercontent.com/assets/7513711/9505507/dce97ce2-4c10-11e5-85b7-8f19dfe2ad46.png)

![screenshot - f12 network tab - response tab](https://cloud.githubusercontent.com/assets/7513711/9505512/e3f5aaa6-4c10-11e5-85c1-2051eee61c82.png)

**Comment by mbressman** [comment]
One other thing I forgot to post (but it was part of the forum thread where I originally posted) - I was advised to try and create another user on ownCloud and see if the search_lucene worked fully with that new user, and it did, including finding text within certain file types (i.e. TXT, PDF, etc.).  So that's why I was thinking that maybe clearing whatever is considered the "search index" under my user account and letting it rebuild might be a possible solution?  Thanks for assisting!! :)

**Comment by MorrisJobke** [comment]
> So that's why I was thinking that maybe clearing whatever is considered the "search index" under my user account and letting it rebuild might be a possible solution? Thanks for assisting!! :)

Good to know :)

@butonic Can you help with the "clearing search index" question?

**Comment by butonic** [comment]
There is still no button to clear the search index. To completely clean it you need to do two things:

1. Delete the `path/to/data/youruser/lucene_index` folder.
2. remove your users files from the `oc_lucene_status` table:
```sql
DELETE FROM oc_lucene_status
WHERE fileid IN (
  SELECT fileid
  FROM oc_storages s
  JOIN oc_filecache fc ON s.numeric_id = fc.storage
  WHERE s.id = 'home::yourusername' -- <- replace yourusername ;)
);
```

**Comment by mbressman** [comment]
Will definitely give this a try and report back.  Also, should I leave the changes I made to those 4 files, or revert them back?  Thanks!

**Comment by mbressman** [comment]
OK, so I ran the SQL query you provided via phpmyadmin, and it actually completely emptied the oc_lucene_status table (but I'm assuming this is correct since I am the only user on the ownCloud, so all of those files in that table would most likely be for my user account, right?), and then I deleted all files within my user account's lucene_index folder.  I also reverted the 4 changed files back to the original's (wasn't sure if I should have kept them or not, and figured it was safer to put them back to the original's that were there from the initial v8.1.1 upgrade).  I then opened up my ownCloud and clicked on the search bar in the upper right, and now it's doing a count-down of file's it's indexing once again (i.e. Indexing...9485 files left, etc.).  So I'm assuming once this completes, search should work again, but I will report back here one way or the other.  Thanks again for all the help!!
